### DISCUSSION TITLE
Controller模式，重启后，发现主从不同步，报错信息：Failed to find a consistent point between masterEpoch:

### DISCUSSION THREAD

**Discussion by onejimmyboy** [apache/rocketmq#7674] [General]
version:5.1.3
集群形态：嵌入式部署Controller，共3节点（1个节点部署Controller+namesrv，另外两个节点都部署namesrv+Controller+broker）。
现象，集群搭建的时候，服务正常，主从同步。之后发送消息，重启服务后，主从无法同步，信息如下：

![image](https://github.com/apache/rocketmq/assets/57432151/68a15a93-8c78-4bd9-8710-3f4c12497f07)
![image](https://github.com/apache/rocketmq/assets/57432151/c06b9139-e16b-4acd-8cd8-a5ac26dcb082)
疑问：报错的信息是Failed to find a consistent point between masterEpoch，
        1）这个报错信息代表的是什么意思？导致报错的原因？
         2）如何修复呐？自己有尝试重启，但是依然无法修复

**Comment by RongtongJin** [comment]
看着主备之间的epochFileCheckpoint中的epoch无法对齐，但这些日志无法查出报错的原因。有办法稳定复现吗？

**Reply by onejimmyboy** [reply]
重新部署了一套，没有复现。不确定是什么问题导致的，不知道有没有其他小伙伴有遇到过？另外如果遇到了这个问题的话，还有办法恢复吗？比如清空拷贝主节点的数据到从节点（不知道是否可行）等

**Comment by lialatdd** [comment]
我这边部署5.1.3版本的controller+broker集群，一主一从的模式下，也出现了这个问题

**Comment by lialatdd** [comment]
![image](https://github.com/user-attachments/assets/1a54702d-1133-4f93-9947-a2c0ee6c3a98)

**Comment by aligege123** [comment]
我这5.2.0也有这个问题☹️

**Comment by iamgd67** [comment]
好像是长时间不同步，会有这个问题，epoch数据会被清理。可以换个数据目录，作为新节点同步。理论上代码里应该handle这种情况
